From: Ben Hutchings <ben@decadent.org.uk>
Date: Tue, 05 Aug 2014 14:00:19 +0100
Subject: mutex: Avoid ABI change in 3.14.14

The dependencies for CONFIG_MUTEX_SPIN_ON_OWNER were changed so
mutex::spin_mlock is no longer defined or used on some architectures.
Change the condition for its definition back to what it was before.

--- a/include/linux/mutex.h
+++ b/include/linux/mutex.h
@@ -54,7 +54,7 @@ struct mutex {
 #if defined(CONFIG_DEBUG_MUTEXES) || defined(CONFIG_SMP)
 	struct task_struct	*owner;
 #endif
-#ifdef CONFIG_MUTEX_SPIN_ON_OWNER
+#if defined(CONFIG_SMP) && !defined(CONFIG_DEBUG_MUTEXES)
 	void			*spin_mlock;	/* Spinner MCS lock */
 #endif
 #ifdef CONFIG_DEBUG_MUTEXES
